app/routes/task_routes.py

```
@@ -1 +1,78 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, Response, request
+from .route_utilities import validate_model, create_model, get_models_with_filters, post_slack_message
+from app.db import db
+from app.models.task import Task
+from datetime import datetime
+
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+
+@tasks_bp.get("")
+def get_all_tasks():
+    return get_models_with_filters(Task, request.args)
+
+
+@tasks_bp.post("")
+def create_task():
+    request_body = request.get_json()
+    return create_model(Task, request_body)
+
+
+@tasks_bp.get("/<task_id>")
+def get_one_task(task_id):
+    task = validate_model(Task, task_id)
+    response_body = {"task": task.to_dict()}
+    
+    return response_body
+
+
+@tasks_bp.put("/<task_id>")
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+
+    request_body = request.get_json()
+    task.title = request_body.get("title", task.title)
+    task.description = request_body.get("description", task.description)
```

If you used square bracket notation to get the value of "title" and "description", but those keys weren't present in `request_body` then your code would throw a KeyError. I'm guessing you use `get` and provide a default value if the key isn't found so that this error doesn't get thrown from this PUT request. Is that right?

However, if `request_body` doesn't have "title" or "description" then you grab these values from `task` which is the record that is retrieved from the database when you invoked `validate_model` on line 32. This means, if the request body of the PUT request doesn't have the right keys then you use `task` from the db to re-assign title and description. 

So if my task that is saved to the DB had title 'Walk **the** dog' and description 'Walk the dog for **30 minutes**' and my incorrect PUT request body is:
```
{
    # incorrect key, should be description
    "title": "Walk my dog"
    "task_description": "Walk the dog for 1 hour"
}
```

Then you just get the title and description my task already has and commit to the DB, which doesn't actually make the update. 

If my request body doesn't have keys needed to fulfill the request, then I'd expect the route to throw an error or give me an error response.

---

Thanks for the explanation. Can I just delete the default value then since the tasks were supposed to all have a description?

---

What is the default value?

If the request body doesn't have what you need to process the request and return a response, I dont think anything should be deleted.

You should return an error so the client knows they need to send back a correct request body